ddl/db_test.go

```diff
@@ -1665,6 +1665,16 @@ func (s *testDBSuite) TestCreateTableWithPartition(c *C) {
 		partition p2 values less than maxvalue
 	);`)
 	c.Assert(ddl.ErrNotAllowedTypeInPartition.Equal(err), IsTrue)
+
+	// TODO: This SQL should return ErrPartitionFunctionIsNotAllowed: ERROR 1564 (HY000): This partition function is not allowed
+	_, err = s.tk.Exec(`create TABLE t9 (
+	col1 int
+	)
+	partition by range( case when col1 > 0 then 10 else 20 end ) (
```

Comment:
This expression is not supported `case when col1 > 0 then 10 else 20 end`.

Reply:
Ye~
The correct behavior is return error: ERROR 1564 (HY000): This partition function is not allowed.
I add a TODO in the comment and create an issue about this. issue#7058.